### ISSUE TITLE
[blaine branch] handling unregistered tasks

### ISSUE THREAD

**Issue by pkozlowski-opensource** [robrich/orchestrator#54]
tl;dr: how new (blaine) orchestrator should report errors to gulp4?

I wonder how a situation where someone tries to invoke a non-existing task should be handled. Currently there is only assert in bach so trying to run a non-existing task will result in something like:

> assert.js:92
>   throw new assert.AssertionError({
>         ^
> AssertionError: Only functions can be combined, got undefined for argument 0
>     at /Users/pawelkozlowski/work/gitrepos/pkozlowski-opensource/gulp4-poc/node_modules/orchestrator/node_modules/bach/lib/verifyArguments.js:8:12

which is correct but not very user-friendly. 

In the current version of the orchestrator the `task_not_found` is emitted and reacted-upon in gulp. Is this the model we want to keep for Gulp4 given that:
- current orchestrator is not even an event emitter (although probably will become one if we want to time tasks and provide their start / stop status)
- there is this comment in gulp: https://github.com/gulpjs/gulp/blob/b4f99943b20bce83468750f9982432ecacc84cc2/bin/gulp.js#L167 

How the "ideal" error reporting from orchestrator to gulp be shaped? Should we keep using events? 
//cc: @contra
